src/app/routines/exersiceModal/ExerciseModal.tsx

```diff
@@ -0,0 +1,150 @@
+import './ExerciseModal.scss';
+import { fetchExerciseByName } from '@/api/exerciseApi';
+import { CreateRoutineExercise } from '@/types/entities/Exercise';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Autocomplete, Button, Dialog, DialogActions, DialogContent, DialogTitle, TextField } from '@mui/material';
+import { validationSchema } from './validationSchema';
+import { Controller, useForm } from 'react-hook-form';
+import { useQuery } from '@tanstack/react-query';
+
+interface ExerciseModalProps {
+  open: boolean;
+  handleClose: () => void;
+}
+
+const ExerciseModal = ({ open, handleClose }: ExerciseModalProps) => {
+  const {
+    register,
+    control,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm({
+    resolver: yupResolver(validationSchema),
+    defaultValues: {
+      exerciseId: '',
+      defaultSets: undefined,
+      defaultReps: undefined,
+      defaultWeight: undefined,
+      defaultRestTime: '',
+      notes: '',
+    },
+  });
+
+  const onSubmit = (data: CreateRoutineExercise) => {
+    console.log(data); // Handle form submission
+    reset();
+    handleClose();
+  };
+
+  const onClose = () => {
+    reset();
+    handleClose();
+  };
+
+  const { data: exerciseOptions } = useQuery({
+    queryFn: async () => fetchExerciseByName(),
+    queryKey: ['exerciseOptions'],
+  });
+
+  return (
+    <Dialog
+      onClose={handleClose}
+      open={open}
+      sx={{
+        '& .MuiPaper-root': {
```

Comment:
minor:
you can add class here and then also apply it with .MuiPaper-root 
Chat will probably easy tell you how to affect that class in css